src/test/java/teammates/test/util/TestUrl.java

```diff
@@ -0,0 +1,16 @@
+package teammates.test.util;
+
+import teammates.common.util.AppUrl;
+import teammates.test.driver.TestProperties;
+
+/**
+ * An extension of {@link AppUrl}, differing in the base URL used for conversion.
+ * In this class, the base URL used is the value of test.app.url in test.properties.
+ */
+public class TestUrl extends AppUrl {
+
+    public TestUrl(String url) {
+        super(TestProperties.inst().TEAMMATES_URL, url);
+    }
+    
+}
```

Comment:
There are actually a few alternatives which I all discarded for various reasons:

1. Move all the specific `withParam(...)` to the base `Url` class itself, so `TestUrl` will inherit directly from `Url` as well. It works, but it'll look strange because not every generic URL should be equipped with such methods (like, why would anyone add a `courseid` param to `http://www.google.com`?).
2. Make `AppUrl` constructor public (so the base URL can be specified), but we're exposing the constructor when there is no need to. Also, the test URL needs to be repeated for every instance, which is really why I opened #4604.
3. Have a factory method `AppUrl.getAppUrlForTest(path)`, but then the main class `AppUrl` have to import `TestProperties.TEAMMATES_URL`, i.e a test class inside production code. If the factory method is in the test package, the constructor has to be public and it goes back to point 2.